rero_ils/modules/loans/serializers.py

```diff
@@ -0,0 +1,115 @@
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Loans serialization."""
+
+from invenio_records_rest.serializers.response import search_responsify
+
+from rero_ils.modules.items.models import ItemCirculationAction
+from rero_ils.modules.loans.api import Loan, LoanState
+from rero_ils.modules.locations.api import Location
+from rero_ils.modules.patrons.api import Patron
+from rero_ils.modules.serializers import JSONSerializer, RecordSchemaJSONV1
+
+from ..documents.api import DocumentsSearch
+from ..items.api import Item
+from ..libraries.api import Library
+
+
+class LoanJSONSerializer(JSONSerializer):
+    """Mixin serializing records as JSON."""
+
+    def post_process_serialize_search(self, results, pid_fetcher):
+        """Post process the search results."""
+        records = results.get('hits', {}).get('hits', {})
+        for record in records:
+            metadata = record.get('metadata', {})
+            # Loan
+            loan = Loan.get_record_by_pid(metadata.get('pid'))
+            # Library name
+            library_pid = metadata.get('library_pid')
+            data = Library.get_record_by_pid(library_pid)
+            metadata['library'] = {
+                'pid': data['pid'],
+                'name': data['name']
+            }
+            del metadata['library_pid']
+            # Document
+            document_pid = metadata.get('document_pid')
+            search = DocumentsSearch().filter(
+                'term', pid=document_pid)
+            document = list(search.scan())[0].to_dict()
+            metadata['document'] = document
+            del metadata['document_pid']
+            # Item
+            item_pid = metadata.get('item_pid', {}).get('value')
+            if item_pid:
+                item = Item.get_record_by_pid(item_pid)
+                metadata['item'] = item.replace_refs().dumps()
+                # Item loan
+                if metadata['state'] == LoanState.ITEM_ON_LOAN:
+                    metadata['overdue'] = loan.is_loan_overdue()
+                # Item request
+                elif metadata['state'] in [
+                    LoanState.PENDING,
+                    LoanState.ITEM_AT_DESK,
+                    LoanState.ITEM_IN_TRANSIT_FOR_PICKUP
+                ]:
+                    self._process_level_1(metadata, item_pid)
+                elif metadata['state'] in [
+                    LoanState.ITEM_RETURNED,
+                    LoanState.ITEM_IN_TRANSIT_TO_HOUSE,
+                    LoanState.CANCELLED
+                ]:
+                    self._process_level_2(metadata, loan)
+                del metadata['item_pid']
+
+        return super(LoanJSONSerializer, self)\
+            .post_process_serialize_search(results, pid_fetcher)
+
+    def _process_level_1(self, metadata, item_pid):
```

Comment:
is it possible to find another name for this method to explain what is doing really? instead of level_1

Reply:
done